crates/sargon/src/gateway_api/methods/state_methods.rs

```diff
@@ -367,6 +367,68 @@ impl GatewayClient {
     }
 }
 
+impl GatewayClient {
+    pub async fn filter_transferable_resources(
+        &self,
+        output: FetchResourcesOutput,
+    ) -> Result<FetchTransferableResourcesOutput> {
+        let mut non_transferable_resources = Vec::new();
```

Comment:
duplicates allowed? If not I prefer `IndexSet` (and then convert to vec if needed by data structures later)

Reply:
unless I am missing something, it is logically impossible for an account to have same resource twice (it would just increase their corresponding `amount`).